### PR TITLE
development CLI

### PR DESCRIPTION
**Pull request by splitbrain**
This CLI is basically a replacement for pluginwiz.dokuwiki.org - It allows the creation of skeleton files for plugins and templates. Unlike the wizard, it can also be used to add additional files to existing plugins.

Skeleton files are downloaded automatically from the Plugin Wizard repository.

Support for templates may not work 100% yet.

### CODE REVIEW

**Comment by splitbrain** [comment]
General question: is it worth to have this in DokuWiki or should it be a plugin?

**Comment by micgro42** [comment]
Ooh!

This should likely be a plugin as only a few people (devs writing plugins) will use it and even those not that often (_Creating_ an initial new plugin is rare. _Maintaining_ it is where the real workload lies.)

But yes, having an actual easy to use API to create plugins would indeed be nice. :)

**Comment by micgro42** [comment]
(A while ago, I started something similar-ish: https://github.com/micgro42/dokuwiki-plugin-dx
It is a plugin to keep existing plugins aligned in order to ease the maintenance burden. To that end, it is intentionally opinionated.
But it is inspired by the plugin wizard and uses a lot of similar technology.
It is different from what is proposed here in that it does not try to cover the initial creation.
It isn't fully done yet because I got distracted again by the next shiny idea I had. \**sigh*\*)

File: bin/dev.php [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        $options->registerCommand('addLang', 'add the language files');
+
+        $options->registerCommand('addComponent', 'add a component');
+        $options->registerArgument('type', 'The type of the component. Needs to be one of ' .
+            join(', ', PluginController::PLUGIN_TYPES),
+            true, 'addComponent'
+        );
+        $options->registerArgument('name', 'Optional name of the component', false, 'addComponent');
+
+        $options->registerCommand('deletedFiles', 'create the list of deleted files base on the git history');
```
Comment (micgro42) [inline bin/dev.php:53]: This will come in handy!

**Comment by splitbrain** [comment]
> Maintaining it is where the real workload lies.

That's part of the reason I wrote this. So far I used it a couple of time on existing plugins to add new tests or a new component without having to manually copying boiler plate from somewhere.

> This should likely be a plugin 

Fair enough. I might merge the features of https://github.com/splitbrain/dokuwiki-versionfix

> I started something similar-ish

Hah! I didn't know! I'll see if I can lift some code from your plugin...

**Comment by splitbrain** [comment]
This feature has now been released as a plugin: https://github.com/splitbrain/dokuwiki-plugin-dev

**Comment by micgro42** [comment]
> This feature has now been released as a plugin: [splitbrain/dokuwiki-plugin-dev](https://github.com/splitbrain/dokuwiki-plugin-dev)

Looking forward to trying it out!

**Pull request closed without merging**